File: challenge/level2/피보나치 수.c
#include <stdio.h>

int solution(int n)
{
    int first = 1, second = 1;
    
    for(int i = 2; i < n; i++)
    {
        int tmp = first;
        first = (first + second) % 1234567;
        second = tmp % 1234567;
    }
    
    return first;
}

File: challenge_Java/피보나치 수.java
class Solution {
    public int solution(int n) {
        int first = 0, second = 1;
        while (n-- > 1) {
            int temp = second;
            second = (first + second) % 1234567;
            first = temp;
        }
        return second;
    }
}
